### PR TITLE
API: websocket logs endpoint for polling envoy fleet logs

### PR DESCRIPTION
**Pull request by kylehodgetts**
Changes:
- update rbac to allow kusk api websocket to poll pod logs
- add kusk-gateway-api service account (installing via kusk assigned the kusk-manager role to the api)
- add websocket container to api deployment
- add websocket port to api service
- add /logs endpoint to api's openapi spec

### CODE REVIEW

**Review by mbana — APPROVED** [review]
LGTM.

**Review by jasmingacic — APPROVED** [review]

**Review by jasmingacic — APPROVED** [review]

File: cmd/kusk/manifests/api_server.yaml [hunk 1]
```diff
@@ -113,7 +122,24 @@ spec:
         - image: kubeshop/kusk-gateway-api:v1.1.10
           imagePullPolicy: Always
           name: kusk-gateway-api
-      serviceAccountName: kusk-gateway-manager
+        - name: kusk-gateway-api-websocket
+          image: kubeshop/kusk-gateway-api-websocket:v.1.1.10
```
Comment (mbana) [inline cmd/kusk/manifests/api_server.yaml:126]: I can't find the tag for the image in https://hub.docker.com/r/kubeshop/kusk-gateway-api-websocket/tags, only `latest` is present. Am I looking at the wrong page?

**Review by mbana — CHANGES_REQUESTED** [review]

**Comment by jasmingacic** [comment]
Should we merge this?

File: cmd/kusk/manifests/api_server.yaml (hunk 1 above)
Comment (kylehodgetts) [reply cmd/kusk/manifests/api_server.yaml:126]: The release CI in github actions will tag it as such. I want to keep it in lockstep with the api @mbana

Comment (kylehodgetts) [reply cmd/kusk/manifests/api_server.yaml:126]: This tag is wrong actually, it should be v1.1.10

File: cmd/kusk/manifests/api_server.yaml [hunk 2]
```diff
@@ -113,7 +122,24 @@ spec:
         - image: kubeshop/kusk-gateway-api:v1.1.10
           imagePullPolicy: Always
           name: kusk-gateway-api
-      serviceAccountName: kusk-gateway-manager
```
Comment (mbana) [inline cmd/kusk/manifests/api_server.yaml:116]: Why did you remove this?

File: cmd/kusk/manifests/api_server.yaml [hunk 3]
```diff
@@ -113,7 +122,24 @@ spec:
...
+        - name: kusk-gateway-api-websocket
+          image: kubeshop/kusk-gateway-api-websocket:v1.1.10
```
Comment (mbana) [inline cmd/kusk/manifests/api_server.yaml:126]: According to <https://hub.docker.com/r/kubeshop/kusk-gateway-api-websocket/tags>, there's no such image tagged `kubeshop/kusk-gateway-api-websocket:v1.1.10`.

**Review by mbana — CHANGES_REQUESTED** [review]
Please install:

`go install github.com/go-bindata/go-bindata/v3/go-bindata@v3.1.3`

And generate `cmd/kusk/cmd/manifest_data.go` again. Also, please see comment about the image not being available on DockerHub.

File: cmd/kusk/manifests/api_server.yaml (hunk 2 above)
Comment (kylehodgetts) [reply cmd/kusk/manifests/api_server.yaml:116]: Its the wrong service account. You can see I define the proper service account on https://github.com/kubeshop/kusk-gateway/pull/946/files#diff-070d573c248f6027cc11182e49ba2213f9abcfb3c113da56a41fe91dbf0bd82eR142

File: cmd/kusk/manifests/api_server.yaml (hunk 3 above)
Comment (kylehodgetts) [reply cmd/kusk/manifests/api_server.yaml:126]: This needs updating to v1.2.1 as per my release today of the api and websocket

File: cmd/kusk/manifests/api_server.yaml (hunk 2 above)
Comment (mbana) [reply cmd/kusk/manifests/api_server.yaml:116]: OK. Cool. Fine.

**Review by mbana — APPROVED** [review]
LGTM.
